cairo/ethereum/cancun/vm/precompiled_contracts/ecrecover.cairo

```diff
@@ -56,11 +59,13 @@ func ecrecover{
 
     let is_v_valid = is_v_27.value + is_v_28.value;
 
-    if (is_v_valid != 0) {
+    if (is_v_valid == 0) {
         tempvar ok = cast(0, EthereumException*);
         return ok;
     }
 
+    tempvar y_parity = U256(new U256Struct(v.value.low - 27, 0));
```

Comment:
There is no check on v in `try_get_point_from_x`. This should be added to be sound.

Reply:
i'm not sure I get it?

Reply:
what should be added?

Reply:
Sorry for not giving more details:
https://github.com/kkrt-labs/keth/blob/1a22cdeae04cb1e5774c4a5e3763069879d70071/python/cairo-ec/src/cairo_ec/ec_ops.cairo#L29

There is no check on v value in `try_get_point_from_x` (itself called by `try_recover_public_key`)
So there is a missing constraint on v that makes this unsound.

Reply:
You right this is another subject. Just want to make sure we do not forget it